ephios/core/migrations/0026_userprofile_disabled_notifications_and_more.py

```diff
@@ -0,0 +1,61 @@
+# Generated by Django 4.2.7 on 2024-01-06 13:21
+
+from django.db import migrations, models
+
+import ephios.extra.json
+from ephios.core.services.notifications.backends import enabled_notification_backends
+from ephios.core.services.notifications.types import notification_type_from_slug
+
+
+def migrate_disabled_notifications(apps, schema_editor):
+    UserProfile = apps.get_model("core", "UserProfile")
+    db_alias = schema_editor.connection.alias
+    backends = [backend.slug for backend in enabled_notification_backends()]
+    for profile in UserProfile.all_objects.using(db_alias).all():
+        preferences = profile.preferences["notifications__notifications"]
+        for notification_type, active_backends in preferences.items():
+            for backend in set(backends) - set(active_backends):
+                profile.disabled_notifications.append([notification_type, backend])
+        profile.save()
+
+
+def revert_disabled_notifications(apps, schema_editor):
+    UserProfile = apps.get_model("core", "UserProfile")
+    db_alias = schema_editor.connection.alias
+    backends = [backend.slug for backend in enabled_notification_backends()]
+    for profile in UserProfile.all_objects.using(db_alias).all():
+        preferences = profile.preferences["notifications__notifications"]
```

Comment:
profile.preferences does not exist during the migration for whatever reason. we need to figure out how to access this